simde/x86/sse4.2.h

```diff
@@ -275,15 +275,22 @@ simde_mm_crc32_u8(uint32_t prevcrc, uint8_t v) {
   #if defined(SIMDE_X86_SSE4_2_NATIVE)
     return _mm_crc32_u8(prevcrc, v);
   #else
-    uint32_t crc = prevcrc;
-    crc ^= v;
-    for(int bit = 0 ; bit < 8 ; bit++) {
-      if (crc & 1)
-        crc = (crc >> 1) ^ UINT32_C(0x82f63b78);
-      else
-        crc = (crc >> 1);
-    }
-    return crc;
+    #if defined(SIMDE_ARM_NEON_A64V8_NATIVE) && defined(__ARM_FEATURE_CRC32)
+      __asm__ __volatile__("crc32cb %w[c], %w[c], %w[v]\n\t"
```

Comment:
I'd like to avoid inline assembly.  Obviously this is GCC-style inline assembly, so it won't work in MSVC (which yes, can target ARM).  The __volatile__ is also a barrier for optimizations.

There are intrinsics for this in `arm_acle.h`.  If the compiler supports ACLE `__ARM_ACLE` should be defined.  However, GCC doesn't yet implement all of ACLE, though it does have the __crc32* intrinsics, so you can do something like

```c
#if defined(__ARM_ACLE) || (defined(__GNUC__) && defined(__ARM_FEATURE_CRC32))
  #include <arm_acle.h>
#endif
```

Then instead of using inline assembly here you can use the `__crc32cb` function.

Reply:
I have tried to do this, but for ```mm_crc32_u64``` __crc32cx is not available do I need to include some other header file for accessing that function?
Thank You

Reply:
fixed this